### DISCUSSION TITLE
Make the wavetrigger play notes

### DISCUSSION THREAD

**Discussion by baloe** [FortySevenEffects/arduino_midi_library#313] [Q&A]
Hello everyone. I am struggling to play a note on the [Wavetrigger board](https://robertsonics.com/wav-trigger/).

I have done the following:

- I have unpacked the MIDI example [MellotronDemo.zip](https://robertsonics.com/resources/wavtrigger/MellotronDemo.zip) onto the SD card
- I have verified that a sound is played after pressing the test button on the Wavetrigger board.
- I have hooked up:
  - Arduino TX (0) with Wavetrigger RX
  - Arduino GND with Wavetrigger GND

I have been playing around with a few arduino sketches, like the one below, but none of them ever produced a sound.
Could anyone give me a hint on how to get this to work?

According to the [manual on `sendNoteOn`](https://fortyseveneffects.github.io/arduino_midi_library/a00032.html#ga580fb85c5b020566dc04cabc42aae9dc) I do have to pass in bytes, but I did also try to use `bytes(note)` in the sketch below.

```
#include <MIDI.h>

// Create and bind the MIDI interface to the default hardware Serial port
//MIDI_CREATE_INSTANCE(HardwareSerial, Serial, MIDISerial);
MIDI_CREATE_DEFAULT_INSTANCE();

void setup()
{ 
    MIDI.begin();
}

void loop()
{
    for (int note = 0; note < 100; note++) {
      MIDI.sendNoteOn(note, 127, 1);
      delay(100);
      MIDI.sendNoteOff(note , 0, 1);
      delay(100);
    }
}
```
(I am also asking in an issue over at the Wavetrigger repository: https://github.com/robertsonics/WAV-Trigger-Arduino-Serial-Library/issues/25)

**Comment by franky47** [comment]
That page suggests MIDI support was added in the latest firmware version of the Wavetrigger, just checking if that's your case before digging any further.

Edit: My bad, it's just realtime controls and pitch bend that were in the latest firmware.

Could this be a channel issue? Maybe try other channels than 1 and see what happens.

**Comment by baloe** [comment]
mh I don't know which firmware version is on the board but I assume it is 1.28 because (if I remember correctly) this was the latest stable version at the time I bought the board (and it still is). Up to version 1.30 [track numbers are computed as](https://robertsonics.com/wav-trigger-online-user-guide/#chapter10)
```
Track Number = (MIDI Channel * 128) + (MIDI Bank Number * 128) + MIDI Note Number
```
That track number then determines the first three 0-leading digits in the file name that is loaded to play the note.
 I currently have files 043-077, so yes, a channel=0 would seem to be more promising, but the documentation for the MIDI library requires 1-16.  I also don't know how to set the MIDI bank number.

**Comment by baloe** [comment]
I finally succeeded in triggering sound, but oddly I have to use channel 2 where I was expecting channel 0 to be appropriate: https://github.com/robertsonics/WAV-Trigger-Arduino-Serial-Library/issues/26
It's really confusing
